python-libraries/data-platform-catalogue/data_platform_catalogue/client/datahub/datahub_client.py

```diff
@@ -146,6 +176,212 @@ def create_domain(
 
         return domain_urn
 
+    def upsert_athena_database(self, metadata: DatabaseMetadata) -> str:
+        """
+        Define a database. Must belong to a domain
+        """
+        metadata_dict = dict(metadata.__dict__)
+        metadata_dict.pop("version")
+        metadata_dict.pop("owner")
+        metadata_dict.pop("tags")
+
+        name = metadata_dict.pop("name")
+        description = metadata_dict.pop("description")
+
+        database_urn = "urn:li:container:" + "".join(name.split())
+
+        domain = metadata_dict.pop("domain")
+        domain_urn = self.graph.get_domain_urn_by_name(domain_name=domain)
+
+        # domains are to be controlled and limited so we dont want to create one
+        # when it doesn't exist
+        domain_exists = self.check_entity_exists_by_urn(domain_urn)
+        if not domain_exists:
+            raise InvalidDomain(
+                f"{domain} does not exist in datahub - please align data to an existing domain"
+            )
+
+        database_domain = DomainsClass(domains=[domain_urn])
+
+        if domain_urn is not None:
+            metadata_event = MetadataChangeProposalWrapper(
+                entityType="container",
+                changeType=ChangeTypeClass.UPSERT,
+                entityUrn=database_urn,
+                aspect=database_domain,
+            )
+            self.graph.emit(metadata_event)
+            logger.info(f"Database {name} associated with domain {domain}")
+
+        database_properties = ContainerPropertiesClass(
+            customProperties={key: str(val) for key, val in metadata_dict.items()},
+            description=description,
+            name=name,
+        )
+        metadata_event = MetadataChangeProposalWrapper(
+            entityType="container",
+            changeType=ChangeTypeClass.UPSERT,
+            entityUrn=database_urn,
+            aspect=database_properties,
+        )
+        self.graph.emit(metadata_event)
+        logger.info(f"Properties updated for Database {name} ")
+
+        # set platform to athena
+        metadata_event = MetadataChangeProposalWrapper(
+            entityUrn=database_urn,
+            aspect=DataPlatformInstance(
+                platform="urn:li:dataPlatform:athena",
```

Comment:
Have you looked into using platform instances here, rather than the predefined ones? https://datahubproject.io/docs/platform-instances/

We could potentially have a distinct platform specifically for AP